2021/09/deploying-dotnet-5-app-iis.md

```diff
@@ -0,0 +1,74 @@
+---
+author: "Juan Pablo Ventoso"
+date: 2021-09-16
+title: "Deploying a .NET 5 app on IIS"
+tags:
+- dotnet
+- csharp
+- iis
+---
+
+![Puzzle](/blog/2021/09/deploying-dotnet-5-app-iis/puzzle-cropped.jpg)
+
+[Photo](https://flic.kr/p/4riBj8) by [Christian Cable](https://www.flickr.com/photos/nexus_icon/)
+
+.NET 5 is been around for a few years now, after being released at the [.NET Conf 2020](/blog/2020/11/dotnet-5-released-net-conf-2020/), containing the best of both worlds: .NET Core, including multi-platform support and several performance improvements, and .NET Framework, including Windows desktop development support with WPF and Windows Forms (UWP is also supported, but not officially yet).
+
+A .NET Core-based project can be published into any platform (as long as we’re not depending on libraries targeted to .NET Framework), allowing us to save costs by hosting in Linux servers and increasing performance for scalability. But most developers are still using Windows with Internet Informacion Services (IIS) as the publishing target: This might be related to the almost 20 years of history of .NET Framework, comparing to the relatively short history of .NET Core, launched in 2016.
+
+### Our .NET project
+
+We won’t review the steps needed to set up a new .NET 5 project, since this time we are only focusing on publishing what we already have developed. But to understand how our application will integrate with IIS and the framework, it’s important to note a fundamental change any .NET 5 project has in comparison with a .NET Framework one:
+
+Since .NET 5 is .NET Core in its foundation, our project output will actually be a console application. If we create a new .NET Core project, no matter which version we are using, we will find a `Program.cs` file in the root with an application entry point in it, that will look somewhat similar to the one below:
+
+![A common .NET Core app entry point](/blog/2021/09/deploying-dotnet-5-app-iis/dotnet-core-entry-point.jpg)
+
+> The [`WebHost`](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/host/web-host?view=aspnetcore-5.0) object will be the one processing requests to the app, as well as setting configurations like the content root, accessing environment variables, and logging.
+
+This application needs to be executed by the `dotnet` process, which comes with any .NET 5 runtime.
+
+### Installing a .NET 5 runtime
+
+The first step we need to do in our destination server, is to prepare the environment to run .NET 5 apps by installing the [.NET 5 Hosting Bundle](https://dotnet.microsoft.com/download/dotnet/thank-you/runtime-aspnetcore-5.0.9-windows-hosting-bundle-installer), a 65 Mb setup file which has all the basic stuff needed ro tun .NET 5 on Windows. Since .NET can also run in Linux/macOS, on the [official downloads page](https://dotnet.microsoft.com/download/dotnet/5.0)) we can find installer/binaries for these operative systems as well.
```

Comment:
s|installer/binaries|installer executables| perhaps? (I'm trying to wean myself off the term "binaries" when "executable" is the more correct term, and also happens to be the norm in the Windows world!)

Reply:
Yeah, I put it that way just because that was the way it was referred on the Microsoft page (https://dotnet.microsoft.com/download/dotnet/5.0). I just changed it to executables, since it's probably more of a Windows-related term.